File: dataset/CoT/OPENAI_gpt-5-2025-08-07/Task3.py
# Chain-of-Through Step 1: Problem understanding - pangram detection ignoring non-letters, case-insensitive.
# Chain-of-Through Step 2: Security requirements - no unsafe eval/input; handle None safely.
# Chain-of-Through Step 3: Secure coding generation - process only 'a'-'z', early exit when all seen.
# Chain-of-Through Step 4: Code review - no external dependencies or unsafe operations.
# Chain-of-Through Step 5: Secure code output - finalized implementation.

def is_pangram(s: str) -> bool:
    if s is None:
        return False
    seen = [False] * 26
    count = 0
    for ch in s:
        c = ch.lower()
        if 'a' <= c <= 'z':
            idx = ord(c) - ord('a')
            if not seen[idx]:
                seen[idx] = True
                count += 1
                if count == 26:
                    return True
    return count == 26

if __name__ == "__main__":
    tests = [
        "The quick brown fox jumps over the lazy dog",
        "Hello, world!",
        "",
        "Sphinx of black quartz, judge my vow 123!!!",
        "abcdefghijklmno pqrstuvwxy"
    ]
    for t in tests:
        print("True" if is_pangram(t) else "False")

File: dataset/CoT - Copy/OPENAI_gpt-5-2025-08-07/Task3.c
/* Chain-of-Through Step 1: Problem understanding - pangram detection, ignore non-letters, case-insensitive.
   Chain-of-Through Step 2: Security requirements - safe character handling, avoid undefined behavior with tolower.
   Chain-of-Through Step 3: Secure coding generation - strict 'a'..'z' checks, early exit.
   Chain-of-Through Step 4: Code review - ensure no buffer overflows and safe iteration.
   Chain-of-Through Step 5: Secure code output - final implementation. */

#include <stdio.h>
#include <ctype.h>
#include <stdbool.h>

bool is_pangram(const char *s) {
    if (s == NULL) return false;
    bool seen[26] = { false };
    int count = 0;
    for (const unsigned char *p = (const unsigned char *)s; *p != '\0'; ++p) {
        char c = (char)tolower(*p);
        if (c >= 'a' && c <= 'z') {
            int idx = c - 'a';
            if (!seen[idx]) {
                seen[idx] = true;
                count++;
                if (count == 26) return true;
            }
        }
    }
    return count == 26;
}

int main(void) {
    const char *tests[5] = {
        "The quick brown fox jumps over the lazy dog",
        "Hello, world!",
        "",
        "Sphinx of black quartz, judge my vow 123!!!",
        "abcdefghijklmno pqrstuvwxy"
    };
    for (int i = 0; i < 5; ++i) {
        bool res = is_pangram(tests[i]);
        printf("%s\n", res ? "True" : "False");
    }
    return 0;
}